### PR TITLE
first attempt at Deque

### PR DESCRIPTION
**Pull request by TreverArnold**

### CODE REVIEW

File: deque.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from doubly_linked_list import DoublyLinkedList
+
+
+class Deque:
+    def __init__(self, iterable=None):
+        self.dll = DoublyLinkedList()
+        self.dll.head = None
+        self.dll.tail = None
```
Comment (W-Ely) [inline deque.py]: Did I miss this in the last data structure?  They look unneeded.

File: deque.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        self.dll._size = 0
```
Comment (W-Ely) [inline deque.py]: Also this.  You tell my whats wrong with accessing `_size` within this class?

File: deque.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            raise ValueError("Pop does not work on an empty queue")
+
+    def popleft(self):
+        try:
+            return self.dll.shift()
+        except ValueError:
+            raise ValueError("Popleft does not work on an empty queue")
+
+    def size(self):
+        return self.dll._size
```
Comment (W-Ely) [inline deque.py]: `._size` tell my why this is an issue.

File: deque.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+from doubly_linked_list import DoublyLinkedList
+
+
+class Deque:
+    def __init__(self, iterable=None):
+        self.dll = DoublyLinkedList()
```
Comment (W-Ely) [inline deque.py]: Why not:
```suggestion
    def __init__(self, iterable=None):
        self.dll = DoublyLinkedList(iterable)
```
?

File: deque.py (hunk 3 above)
Comment (TreverArnold) [reply deque.py]: Its using a private variable from another class.

File: deque.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        try:
+            return self.dll.shift()
+        except ValueError:
+            raise ValueError("Popleft does not work on an empty queue")
+
+    def size(self):
+        return len(self.dll)
+
+    def __len__(self):
+        return self.dll._size
```
Comment (W-Ely) [inline deque.py]: Same, _size

Comment (TreverArnold) [reply deque.py]: I thought I had changed that one too, must have slipped by

**Review by W-Ely — APPROVED** [review]

**Pull request merged**